app/questionnaire/questionnaire_store_updater.py

```diff
@@ -95,21 +89,27 @@ def remove_list_item_and_answers(self, list_name: str, list_item_id: str):
             list_item_id=list_item_id
         )
 
-    def add_completed_location(self, location=None, section_id=None):
+        self._progress_store.remove_progress_for_list_item_id(list_item_id=list_item_id)
+
+    def add_completed_location(self, location=None):
         location = location or self._current_location
-        section_id = section_id or self._current_section_id
 
-        self._progress_store.add_completed_location(section_id, location)
+        self._progress_store.add_completed_location(location)
 
     def remove_completed_location(self, location=None):
         location = location or self._current_location
-        self._progress_store.remove_completed_location(
-            self._current_section_id, location
-        )
+        self._progress_store.remove_completed_location(location)
+
+    def update_section_status(self, section_status, location=None):
+        if location:
```

Comment:
This could be simplified to:
```suggestion
        location = location or self._current_location
        
        self._progress_store.update_section_status(
            section_status, section_id=location.section_id, list_item_id=location.list_item_id
        )
```

Reply:
Done